core/src/main/java/org/apache/iceberg/SnapshotParser.java

```diff
@@ -76,18 +76,8 @@ static void toJson(Snapshot snapshot, JsonGenerator generator) throws IOExceptio
       generator.writeEndObject();
     }
 
-    String manifestList = snapshot.manifestListLocation();
-    if (manifestList != null) {
-      // write just the location. manifests should not be embedded in JSON along with a list
-      generator.writeStringField(MANIFEST_LIST, manifestList);
-    } else {
-      // embed the manifest list in the JSON, v1 only
-      generator.writeArrayFieldStart(MANIFESTS);
-      for (ManifestFile file : snapshot.allManifests(DUMMY_FILE_IO)) {
-        generator.writeString(file.path());
-      }
-      generator.writeEndArray();
-    }
+    // write just the location. manifests should not be embedded in JSON along with a list
+    generator.writeStringField(MANIFEST_LIST, snapshot.manifestListLocation());
```

Comment:
I don't think that we can make this change. We can ensure that all of the snapshots have a manifest list when producing them, but unless we want to change existing snapshots, we can't change the snapshot serialization. If there is not manifest list, we have to write out the manifests.

Reply:
ok good point, I added that piece back